### PR TITLE
Move the deploy test action to a custom flowzone test

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: patch

### CODE REVIEW

**Comment by klutchell** [comment]
@thgreasi I was wrong, since we are looking for the published docker image this should actually be a custom publish step that runs after docker publish.

**Comment by thgreasi** [comment]
Fine @klutchell let me try it

File: .github/actions/publish/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        # Use the Docker image built from the PR
+        yq e -i '.runs.image = "docker://ghcr.io/balena-io/deploy-to-balena-action:${{ github.event.pull_request.head.sha }}"' \
+          action.yml
+        # Overwrite the Dockerfile with minimal number of steps so the test deploy finishes quickly
+        echo FROM busybox > Dockerfile
+
+    - name: Deploy to a test fleet
+      uses: ./
+      with:
+        balena_token: ${{ inputs.secrets.BALENA_API_KEY }}
```
Comment (klutchell) [inline .github/actions/publish/action.yml:27]:
```suggestion
        balena_token: ${{ fromJSON(inputs.secrets).BALENA_API_KEY }}
```

Comment (thgreasi) [reply .github/actions/publish/action.yml]: right...

**Comment by klutchell** [comment]
Hmm, it looks like custom publish runs at the same time as docker publish in Flowzone, so the image is not available yet.

We could change the Flowzone behaviour, or add a wait loop to keep inspecting the manifest until it is available.

**Comment by klutchell** [comment]
LGTM

**Comment by thgreasi** [comment]
you are awesome @klutchell :clap:

**Pull request merged**